app/_src/kubernetes-ingress-controller/references/custom-resources-2.7.x.md

```diff
@@ -0,0 +1,464 @@
+| `hash_on_header` _string_ | HashOnHeader defines the header name to take the value from as hash input. Only required when "hash_on" is set to "header". |
+| `hash_fallback_header` _string_ | HashFallbackHeader is the header name to take the value from as hash input. Only required when "hash_fallback" is set to "header". |
+| `hash_on_cookie` _string_ | The cookie name to take the value from as hash input. Only required when "hash_on" or "hash_fallback" is set to "cookie". |
+| `hash_on_cookie_path` _string_ | The cookie path to set in the response headers. Only required when "hash_on" or "hash_fallback" is set to "cookie". |
+| `hash_on_query_arg` _string_ | HashOnQueryArg is the query string parameter whose value is the hash input when "hash_on" is set to "query_arg". |
+| `hash_fallback_query_arg` _string_ | HashFallbackQueryArg is the "hash_fallback" version of HashOnQueryArg. |
+| `hash_on_uri_capture` _string_ | HashOnURICapture is the name of the capture group whose value is the hash input when "hash_on" is set to "uri_capture". |
+| `hash_fallback_uri_capture` _string_ | HashFallbackURICapture is the "hash_fallback" version of HashOnURICapture. |
+
+
+_Appears in:_
+- [KongIngress](#kongingress)
+
+### KongProtocol
+
+_Underlying type:_ `string`
+
+
+
+
+
+
+
+_Appears in:_
+- [KongClusterPlugin](#kongclusterplugin)
+- [KongIngressRoute](#kongingressroute)
+- [KongPlugin](#kongplugin)
+
+### NamespacedConfigSource
+
+
+
+NamespacedConfigSource is a wrapper around NamespacedSecretValueFromSource.
+
+
+
+| Field | Description |
+| --- | --- |
+| `secretKeyRef` _[NamespacedSecretValueFromSource](#namespacedsecretvaluefromsource)_ | Specifies a name, a namespace, and a key of a secret to refer to. |
+
+
+_Appears in:_
+- [KongClusterPlugin](#kongclusterplugin)
+
+### NamespacedSecretValueFromSource
+
+
+
+NamespacedSecretValueFromSource represents the source of a secret value specifying the secret namespace.
+
+
+
+| Field | Description |
+| --- | --- |
+| `namespace` _string_ | The namespace containing the secret. |
+| `name` _string_ | The secret containing the key. |
+| `key` _string_ | The key containing the value. |
+
+
+_Appears in:_
+- [NamespacedConfigSource](#namespacedconfigsource)
+
+### SecretValueFromSource
+
+
+
+SecretValueFromSource represents the source of a secret value.
+
+
+
+| Field | Description |
+| --- | --- |
+| `name` _string_ | The secret containing the key. |
+| `key` _string_ | The key containing the value. |
+
+
+_Appears in:_
+- [ConfigSource](#configsource)
+
+
+## configuration.konghq.com/v1alpha1
+
+Package v1alpha1 contains API Schema definitions for the configuration.konghq.com v1alpha1 API group.
+
+- [IngressClassParameters](#ingressclassparameters)
+
+### IngressClassParameters
+
+
+
+IngressClassParameters is the Schema for the IngressClassParameters API.
+
+{% include_cached md/kic/crd-ref/ingress_class_parameters_description.md kong_version=page.kong_version %}
```

Comment:
@czeslavo Looks like the include file here is empty. I'm not sure if it should have text in it or if we can just remove this include file and link.

Reply:
It's added here just because we do generate these hooks for every CRD unconditionally, and in the case of `IngressClassParameters` we did not have any note to add, therefore the file is empty. If we remove the file, the include will fail. It'd be cool if we had a `include_cached_if_exists` for such scenarios.

Will it be ok if we add a comment to the file that it's a placeholder and we intentionally leave it empty for now? 
 34a93c8